samcli/local/docker/manager.py

```diff
@@ -33,6 +38,62 @@ def __init__(self,
         self.docker_network_id = docker_network_id
         self.docker_client = docker_client or docker.from_env()
 
+    @staticmethod
+    def is_valid_container_name(name):
+        """
+        Checks if a given name is a valid Docker container name
+
+        Parameters
+        ----------
+        name str
+            Docker container name to check
+
+        Returns
+        -------
+        bool
+            True, if name is a valid Docker container name, False otherwise
+        """
+
+        return True if ContainerManager._container_name_regex.match(name) else False
+
+    @property
+    def is_docker_reachable(self):
+        """
+        Checks if Docker daemon is running. This is required for us to invoke the function locally
+
+        Returns
+        -------
+        bool
+            True, if Docker is available, False otherwise
+        """
+
+        try:
+            self.docker_client.ping()
+            return True
+
+        # When Docker is not installed, a request.exceptions.ConnectionError is thrown.
+        except (docker.errors.APIError, requests.exceptions.ConnectionError):
```

Comment:
Can you add a `LOG.debug(exc_info=ex)` here so we know why it failed when `--debug` is enabled?

Reply:
No problem